mvlearn/decomposition/baseica.py

```diff
@@ -0,0 +1,204 @@
+        instance, random_state is the random number generator; If
+        None, the random number generator is the RandomState instance
+        used by np.random.
+
+    verbose : bool, default=False
+        Print information
+
+    Attributes
+    ----------
+    preproc : ViewTransformer-like instance
+        The fitted instance used for preprocessing
+
+    mixing_ : array, shape (n_views, n_components, n_components)
+        The square mixing matrices, linking preprocessed data
+        and the independent components.
+
+    pca_components_: array, shape (n_components, n_features)
+        Principal axes in feature space, representing the directions
+        of maximum variance in the data. Only used if preproc == "pca".
+
+    components_ : array, shape (n_views, n_components, n_components)
+        The square unmixing matrices
+
+    individual_components_ : list of array
+        Individual unmixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_components, n_features) where n_features is the number of
+        features in the dataset `i`.
+
+    individual_mixing_ : list of array
+        Individual mixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_features, n_components) where n_features is the number of
+        features in the dataset `i`.
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        preproc="pca",
+        random_state=0,
+        verbose=False,
+        multiview_output=True,
+    ):
+        self.verbose = verbose
+        self.n_components = n_components
+        if preproc is None or preproc == "pca" and n_components is None:
+            self.preproc = None
+            self.preproc_name = "None"
+        elif preproc == "pca":
+            self.preproc = ViewTransformer(PCA(n_components=n_components))
+            self.preproc_name = "pca"
+        else:
+            if hasattr(preproc, "transform") and hasattr(
+                preproc, "inverse_transform"
+            ):
+                self.preproc = preproc
+                self.preproc_name = "custom"
+            else:
+                raise ValueError(
+                    "The dimension reduction instance needs"
+                    "a transform and inverse transform method"
+                )
+        self.random_state = random_state
+        self.multiview_output = multiview_output
+
+    def fit(self, X, y=None):
+        """
+        Fits the model
+        Parameters
+        ----------
+        X: list of np arrays of shape (n_voxels, n_samples)
+            Input data: X[i] is the data of subject i
+        """
+        if self.preproc is not None:
+            reduced_X = self.preproc.fit_transform(X)
+        else:
+            reduced_X = X.copy()
+        reduced_X = np.array(reduced_X)
+        unmixings_, S = self.fit_(np.swapaxes(reduced_X, 1, 2))
```

Comment:
private method start with _ and not end with _ so it should be:
```suggestion
        unmixings_, S = self._fit(np.swapaxes(reduced_X, 1, 2))
```